app/build.gradle.kts

```
@@ -190,6 +191,221 @@ knit {
+                // Exclude modules which are not Android libraries
+                // See https://github.com/Kotlin/kotlinx-kover/issues/312
+                ":appconfig",
+                ":libraries:core",
+                ":libraries:coroutines",
+                ":libraries:di",
+                ":libraries:rustsdk",
+                ":libraries:textcomposer:lib",
+            )
+        }
+        .forEach {
+            // println("Add $it to kover")
+            kover(project(it))
+        }
+}
+
+val ciBuildProperty = "ci-build"
+val isCiBuild = if (project.hasProperty(ciBuildProperty)) {
+    val raw = project.property(ciBuildProperty) as? String
+    raw?.toBooleanLenient() == true || raw?.toIntOrNull() == 1
+} else {
+    false
+}
+
+kover {
+    /*
+    // When running on the CI, run only debug test variants
+    if (isCiBuild) {
+        excludeTests {
+            // Disable instrumentation for debug test tasks
+            tasks(
+                "testDebugUnitTest",
+            )
+        }
+    }
+     */
+}
+
+// https://kotlin.github.io/kotlinx-kover/
+// Run `./gradlew :app:koverHtmlReport` to get report at ./app/build/reports/kover
+// Run `./gradlew :app:koverXmlReport` to get XML report
+koverReport {
+    filters {
+        excludes {
+            classes(
+                // Exclude generated classes.
+                "*_ModuleKt",
+                "anvil.hint.binding.io.element.*",
+                "anvil.hint.merge.*",
+                "anvil.hint.multibinding.io.element.*",
+                "anvil.module.*",
+                "com.airbnb.android.showkase*",
+                "io.element.android.libraries.designsystem.showkase.*",
+                "io.element.android.x.di.DaggerAppComponent*",
+                "*_Factory",
+                "*_Factory_Impl",
+                "*_Factory$*",
+                "*_Module",
+                "*_Module$*",
+                "*Module_Provides*",
+                "Dagger*Component*",
+                "*ComposableSingletons$*",
+                "*_AssistedFactory_Impl*",
+                "*BuildConfig",
+                // Generated by Showkase
+                "*Ioelementandroid*PreviewKt$*",
+                "*Ioelementandroid*PreviewKt",
+                // Other
+                // We do not cover Nodes (normally covered by maestro, but code coverage is not computed with maestro)
+                "*Node",
+                "*Node$*",
+                "*Presenter\$present\$*",
+                // Forked from compose
+                "io.element.android.libraries.designsystem.theme.components.bottomsheet.*",
+            )
+            annotatedBy(
+                "io.element.android.libraries.designsystem.preview.PreviewsDayNight",
+                "io.element.android.libraries.designsystem.preview.PreviewWithLargeHeight",
+            )
+        }
+    }
+
+    defaults {
+        // add reports of 'release' Android build variant to default reports
+        mergeWith("release")
```

This seems to fix the wrong coverage, it wasn't taking into account any test result from the debug runs or screenshot tests:

```suggestion
        // add reports of both 'debug' and 'release' Android build variant to default reports
        mergeWith("debug")
        mergeWith("release")
```